src/main/java/com/google/devtools/build/lib/remote/util/Utils.java

```diff
@@ -381,16 +382,18 @@ public static String grpcAwareErrorMessage(IOException e) {
   }
 
   @SuppressWarnings("ProtoParseWithRegistry")
-  public static ListenableFuture<ActionResult> downloadAsActionResult(
+  public static ListenableFuture<CachedActionResult> downloadAsActionResult(
```

Comment:
Please keep this method returning `ListenableFuture<ActionResult>` and construct `CachedActionResult` at the call sites so we can avoid passing magic strings e.g. "disk", "remote".